### DISCUSSION TITLE
Paperless-ngx 2.12.1 Update Fails w/ Error in requirements.txt

### DISCUSSION THREAD

**Discussion by rlkollman** [tteck/Proxmox#3765] [Help Needed] [answered]
While attempting to upgrade Paperless-ngx from 2.11.0 to 2.12.1, the install script fails with this line:
`[ERROR] in line 81: exit code 0: while executing command pip install -r requirements.txt &> /dev/null`

Line 81 of requirements appears to be 
`psycopg-c==3.2.1`

The line above is `psycopg[c]==3.2.1; python_version >= '3.8'`, which obviously appears to be related.

I didn't find any issues on paperless' git related to psycopg, and I'm also not super familiar with python.  I tried `pip install psycopg-c` and received `error: externally-managed-environment`, so I'm not sure about the best way to remedy this situation. Looking at install/paperless-ngx-install.sh, the line before `pip install requirements.txt` should update pip, but when I tried the `pip install psycopg-c`, I am also seeing a notice that there's a new release of pip, so it appears that may not have successfully executed as well?

Any assistance would be appreciated!

Thanks :)

**Comment by tteck** [comment, marked as answer]
https://github.com/tteck/Proxmox/discussions/3300

**Reply by rlkollman** [reply]
Thank you!  For anyone else in this boat, I followed the instructions in the comment and received the same error when running update.  I followed the instructions again, then rebooted the LXC, then ran `update` again, and this time if worked.  For reference, I had manually updated to 12.7 from 12.6 prior to trying this update.
